README.md

```diff
@@ -162,37 +163,396 @@ fun main(args: Array<String>) {
+        DeclareTransactionV2Payload declareTransactionPayloadForFeeEstimate = account.signDeclare(contractDefinition, casmContractDefinition, new ExecutionParams(nonce, new Felt(1000000000000000L)), false);
+        Request<List<EstimateFeeResponse>> feeEstimateRequest = provider.getEstimateFee(List.of(declareTransactionPayloadForFeeEstimate));
+        Felt feeEstimate = feeEstimateRequest.send().get(0).getOverallFee();
+        // Make sure to prefund the account with enough funds to cover the fee for declare transaction
+
+        // Declare a contract
+        ExecutionParams params = new ExecutionParams(nonce, new Felt(feeEstimate.getValue().multiply(BigInteger.TEN)));
+        DeclareTransactionV2Payload declareTransactionPayload = account.signDeclare(contractDefinition, casmContractDefinition, params, false);
+
+        Request<DeclareResponse> request = provider.declareContract(declareTransactionPayload);
+        DeclareResponse response = request.send();
+        Felt hash = response.getTransactionHash(); // Hash of the transaction that declares a contract
+    }
+}
+```
+or in Kotlin
+
+```kotlin
+fun main(args: Array<String>) {
+    // Create a provider for interacting with StarkNet
+    val provider = JsonRpcProvider("https://example-node-url.com/rpc", StarknetChainId.TESTNET)
+
+    // Set up an account
+    val privateKey = Felt.fromHex("0x1234")
+    val accountAddress = Felt.fromHex("0x1236789")
+    // ⚠️ WARNING ⚠️ Both the account address and private key are for demonstration purposes only.
+    val account = StandardAccount(accountAddress, privateKey, provider)
+
+    // Import a compiled contract
+    val contractCode = Path.of("contract.json").readText()
+    val casmCode = Path.of("contract.casm").readText()
+    val contractDefinition = Cairo1ContractDefinition(contractCode)
+    val casmContractDefinition = CasmContractDefinition(casmCode)
+    val nonce = account.getNonce().send()
+
+    // Estimate fee for declaring a contract
+    val declareTransactionPayloadForFeeEstimate = account.signDeclare(
+        sierraContractDefinition = contractDefinition,
+        casmContractDefinition = casmContractDefinition,
+        params = ExecutionParams(nonce, Felt(1000000000000000L)),
+    )
+    val feeEstimateRequest = provider.getEstimateFee(listOf(declareTransactionPayloadForFeeEstimate))
+    val feeEstimate = feeEstimateRequest.send().first().overallFee
+    // Make sure to prefund the account with enough funds to cover the fee for declare transaction
+
+    // Declare a contract
+    val params = ExecutionParams(nonce, Felt(feeEstimate.value.multiply(BigInteger.TEN)))
+    val declareTransactionPayload = account.signDeclare(
+        sierraContractDefinition = contractDefinition,
+        casmContractDefinition = casmContractDefinition,
+        params = params,
+    )
+
+    val request = provider.declareContract(declareTransactionPayload)
+    val response = request.send()
```

Comment:
Just a detail, since it is in the other code example:
```suggestion
    val request = provider.declareContract(declareTransactionPayload)
    val response = request.send()
    val hash = response.transactionHash // Hash of the transaction that declares a contract
```